textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/DataTuple.java

```diff
@@ -29,6 +29,8 @@ public IField getField(int index) {
 
     @Override
     public IField getField(String fieldName) {
+        //TODO the way we are storing the fields and schema, we need to linearly scan to get the field by name
+        //Do we need to change the Tuple design to support constant retrieval?

```

Comment:
To make one map per tuple is too expensive. You can main one map per dataset from FieldName to FieldID in the schema object. Then you can use this map to get the id via schema, and operator the array point access within the tuple.